### PR TITLE
add positionFixed property, making popper with position: 'Fixed' as an option

### PR DESCRIPTION
**Pull request by MrFrankel**

### CODE REVIEW

File: packages/popper/src/modifiers/applyStyle.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export function applyStyleOnLoad(
 
   // Apply `position` to popper before anything else because
   // without the position applied we can't guarantee correct computations
-  setStyles(popper, { position: 'absolute' });
+  setStyles(popper, { position: options.positionFixed ? 'fixed' : 'absolute' });
```
Comment (FezVrasta) [inline packages/popper/src/modifiers/applyStyle.js]: May you move this logic to the `computeStyles` modifier please?

File: packages/popper/tests/functional/core.js [hunk 2]
```diff
@@ -1366,6 +1366,33 @@ describe('[core]', () => {
     });
   });
 
+  it('checks that positionFixed works correctly', done => {
```
Comment (FezVrasta) [inline packages/popper/tests/functional/core.js]: I fear we'll need way more tests than just this 😞

File: packages/popper/src/modifiers/applyStyle.js (hunk 1 above)
Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: will that not cause computation issues? it specifically says in the comment to apply the position ASAP

File: packages/popper/tests/functional/core.js (hunk 2 above)
Comment (MrFrankel) [reply packages/popper/tests/functional/core.js]: Yea, For sure, just want'd to make sure this PR is relevant first. will add more

Comment (FezVrasta) [reply packages/popper/tests/functional/core.js]: Ideally the whole test suite (or at least the ones in core) should run again but with this option set to true, to make sure all the edge cases are supported.

Comment (MrFrankel) [reply packages/popper/tests/functional/core.js]: If the rest of the code seems good to you all add more test.
btw, are you ok with the name, 'positionFixed'?

Comment (MrFrankel) [reply packages/popper/tests/functional/core.js]: @FezVrasta Is there a way to pass params to the spec files in the test? I wouldnt want to copy/paste all the tests with positionFixed: true set. it would be better if we can run all test twice, once with the flag on, once without.

That way all future test will apply too.

Comment (FezVrasta) [reply packages/popper/tests/functional/core.js]: I think you may wrap the tests inside a[false, true].forEach(positionFixed =>

Comment (MrFrankel) [reply packages/popper/tests/functional/core.js]: works like a charm, thanks

File: packages/popper/src/modifiers/applyStyle.js (hunk 1 above)
Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: will do

**Comment by MrFrankel** [comment]
seems some IE11 test are failing, arggg.... Ill fix them... damn you IE!

**Comment by MrFrankel** [comment]
There seems to be an issue with the test, all are failing due to a some type error in IE.
i cant reproduce it even on Saucelabs directly reproducing the test.

@FezVrasta Is there a way I can run the tests locally for IE with sacuelabs?

**Comment by FezVrasta** [comment]
I think the easiest way is to download a free virtual machine from modern.ie and use it to run the tests.

You can copy the karma runner URL and open it in the VM browser and everything should work.

**Comment by MrFrankel** [comment]
copied the karma URL to browser stack, its an IE issue, not sure why my commit raised it. possibly because we run them twice but im not sure.

https://connect.microsoft.com/IE/feedback/details/829392/calling-getboundingclientrect-on-an-html-element-that-has-not-been-added-to-the-dom-causes-unspecified-error

File: packages/popper/src/utils/getBoundingClientRect.js [hunk 3]
```diff
@@ -18,19 +18,21 @@ export default function getBoundingClientRect(element) {
...
       rect.top += scrollTop;
       rect.left += scrollLeft;
       rect.bottom += scrollTop;
       rect.right += scrollLeft;
-    } catch (err) {}
-  } else {
-    rect = element.getBoundingClientRect();
+    }
+    else {
+      rect = element.getBoundingClientRect();
```
Comment (FezVrasta) [inline packages/popper/src/utils/getBoundingClientRect.js]: So is this going to possibly throw error on IE11?

Comment (MrFrankel) [reply packages/popper/src/utils/getBoundingClientRect.js]: No, I wrapped both cases in try catch, before the else was not wrapped so it throws an error in ie11.

Comment (FezVrasta) [reply packages/popper/src/utils/getBoundingClientRect.js]: Yes that was the question, this means that in some cases this will throw an error in IE11 (that will then be caught by the `try...catch`

Comment (MrFrankel) [reply packages/popper/src/utils/getBoundingClientRect.js]: yes, Its a bug in IE apparently which they refuse to fix. I saw that there was I try catch on the IE10 case so I figure wrapping the while thing will led to the same result.

Not sure why this arose now, it seems unrelated to my changes. 
I can only assume its related to running the suites twice.

File: packages/popper/src/modifiers/applyStyle.js (hunk 1 above)
Comment (FezVrasta) [reply packages/popper/src/modifiers/applyStyle.js]: I think this is the latest point needed to finish this PR?

Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: Ive added the case in applyStyles in a57d15560721d115993f673fec923c7d7e4db991.
Did I miss somthing?

Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: @FezVrasta Ive added the case in applyStyles in a57d155.
Did I miss something?

Comment (FezVrasta) [reply packages/popper/src/modifiers/applyStyle.js]: We must export the position value in the `computeStyles` modifier to make sure that people that don't rely on `applyStyle` will not miss this piece of information.

Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: in line 59 of update.js we set the position property according to the positionFixed flag ```data.offsets.popper.position = this.options.positionFixed ? 'fixed' : 'absolute';```

which is later applied in the computeStyle modifier,

```// Styles
  const styles = {
    position: popper.position,
  };
```

So it is exposed to them in the modifier, is this not enough?

Comment (FezVrasta) [reply packages/popper/src/modifiers/applyStyle.js]: I can't seem to see any edit in the `computStyle` modifier, may you add a commit on the interested line?

Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: I haven't edited the computedStyle modifier, it is passed the position via data.offsets.popper.position and through options.positionFixed.

Is that not exposed enough? or am I missing somthing

Comment (FezVrasta) [reply packages/popper/src/modifiers/applyStyle.js]: Uh sorry, I didn't realize we did already handled this case, ok so that's perfect. thanjs.

Comment (MrFrankel) [reply packages/popper/src/modifiers/applyStyle.js]: Yes, Id like to run in my library for a bit to see how it works in the real world before adding the docs to it.

**Comment by FezVrasta** [comment]
May you please rebase on top of master so that I can merge this?

I'm going to release these changes as popper.js@next so that who is interested can play with this for a while before we release it officially

**Comment by MrFrankel** [comment]
https://en.wiktionary.org/wiki/awsome

**Comment by FezVrasta** [comment]
You can install it with `popper.js@next` now
